bundles/io/org.eclipse.smarthome.io.voice/src/main/java/org/eclipse/smarthome/io/voice/text/AbstractRuleBasedInterpreter.java

```diff
@@ -476,40 +568,19 @@ protected String executeSingle(ResourceBundle language, String[] nameFragments,
     /**
      * Tokenizes text. Filters out all unsupported punctuation. Tokens will be lower case.
      *
+     * @param locale the locale that should be used for lower casing
      * @param text the text that should be tokenized
      * @return resulting tokens
      */
-    protected ArrayList<String> tokenize(String text) {
-        ArrayList<String> parts = new ArrayList<String>();
-        String[] split = text.toLowerCase().replaceAll("[^\\w\\s]", " ").split("\\s");
-        for (int i = 0; i < split.length; i++) {
-            String part = split[i].trim();
-            if (part.length() > 0) {
-                parts.add(part);
-            }
-        }
-        return parts;
-    }
-
-    /**
-     * Splits an item's name into single words. It splits whitespace, Pascal, Camel and Snake-casing.
-     *
-     * @param name the name that's to be split
-     * @param toLowerCase if {true}, all resulting fragments will be made lower case
-     * @return resulting fragments of the name
-     */
-    protected ArrayList<String> splitName(String name, boolean toLowerCase) {
+    protected ArrayList<String> tokenize(Locale locale, String text) {
```

Comment:
Yes. So far I am not even sure, if the given tokenization is the right/complete one (see last comment). Therefore there is no testing yet.